File: servoReset.cpp
/*Reset Servos to 50%*/

#include <iostream>
#include <stdlib.h>
#include <string>
#include <pigpio.h>

#define SERVO_PIN 15
#define CENTER_PULSE 1500

int main(int argc, char *argv[]){
	
	unsigned int pulse = 1010;

	while(pulse != 0){

	std::cout << "number between 1000 and 2000, 0 to exit: ";
	std::cin >> pulse;
	
	if(pulse > 2000){
		std::cout << "too high" << std::endl;
		pulse = 1010;
	}else if(pulse == 0){
		break;
	}else if (pulse < 1010){
		std::cout << "too low" << std::endl;
		pulse = 1010;
	}else{	
	if(gpioInitialise() >= 0){
		
		std::cout << "init OK" << std :: endl << gpioGetMode(SERVO_PIN) << std::endl;
		
		if(gpioGetMode(SERVO_PIN) != PI_OUTPUT){
			gpioSetMode(SERVO_PIN, PI_OUTPUT);
		}
		gpioSetPWMfrequency(SERVO_PIN, 50);
		gpioSetPWMrange(SERVO_PIN, 20000);
		
		gpioServo(SERVO_PIN, pulse);
		gpioDelay(1000000);
		
		/*while (gpioGetServoPulsewidth(SERVO_PIN) != CENTER_PULSE){
			gpioServo(SERVO_PIN, CENTER_PULSE);	//Center Servo
			gpioDelay(100);
		}*/
	}
	}
	}
	
	gpioTerminate();
	std::cout << "Terminated" << std::endl;
	
	return 0;
}

File: servoReset.py
#servoReset.py
# Author: Jesse Offei-Nkansah
# Date: 4.9.2018
# Test pulsing to the servo

import pigpio
import time

SERVO_PIN = 15
CENTER_PULSE = 1500

pulse = 1010
pi = pigpio.pi()
print("initializing pi")

while(pulse != 0):
    
    
    
    pulse = (int(input("number between 1010 and 2000, 0 to exit: ")))
    
    if(pulse > 2000):
        print("too high")
        pulse = 1010
    elif(pulse == 0):
        break
    elif(pulse < 1010):
        print("too low")
        pulse = 1010
    else:
        if (not pi.connected):
            print("Pi not connected")
            exit()
        print("Init OK")
        print(pi.get_mode(SERVO_PIN))
        
        if (pi.get_mode(SERVO_PIN) != pigpio.OUTPUT):
            pi.set_mode(SERVO_PIN, pigpio.OUTPUT)
            
        pi.set_servo_pulsewidth(SERVO_PIN, pulse)
        time.sleep(1)
        
while (pi.connected):
    pi.stop()
    
print("disconnected")
